### PR TITLE
feat: `Series.to_dummies`, arrow `sort` and `unique`

### LINKED ISSUE #575: Series.to_dummies
Bit of a tricky one, but it's one that Formulaic use

Polars has `Series.to_dummies`, pandas `pd.get_dummies`, no big deal

🤔 need to figure out how to do it for PyArrow

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #372 
- Closes #575  

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/_arrow/series.py [hunk 1]
```diff
@@ -444,6 +444,39 @@ def is_sorted(self: Self, *, descending: bool = False) -> bool:
...
+    def unique(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._from_native_series(pc.unique(self._native_series))
+
+    def sort(self: Self, *, descending: bool = False) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        series = self._native_series
+        order = "descending" if descending else "ascending"
+        sorted_indices = pc.array_sort_indices(
+            series, order=order, null_placement="at_start"
```
Comment (FBruzzesi) [inline narwhals/_arrow/series.py:456]: with `null_placement="at_start"` I am emulating what polars does as default since we don't have such option for now

**Comment by MarcoGorelli** [comment]
this is literal 🔥 🔥 🔥 

looks like there's just an issue with old versions of Polars not having `drop_first`

**Comment by FBruzzesi** [comment]
Damn! I could check only until pandas 1.0, not even the docs get so far in the past!
I will need to come back to this a bit later today

**Comment by MarcoGorelli** [comment]
> pandas 1.0

Polars, not pandas 😄

File: narwhals/_polars/series.py [hunk 2]
```diff
@@ -168,6 +169,21 @@ def __rpow__(self, other: PolarsSeries | Any) -> Self:
     def __invert__(self) -> Self:
         return self._from_native_series(self._native_series.__invert__())
 
+    def to_dummies(
+        self: Self, *, separator: str = "_", drop_first: bool = False
+    ) -> PolarsDataFrame:
+        from narwhals._polars.dataframe import PolarsDataFrame
+
+        if self._backend_version < (0, 20, 15):
```
Comment (FBruzzesi) [inline narwhals/_polars/series.py:177]: I am bisecting the sh*t out of it to find out the first version :sweat_smile:

File: narwhals/_arrow/series.py (hunk 1 above)
Comment (EdAbati) [reply narwhals/_arrow/series.py:480]: should we add this in the code as comment? (I didn't know that this is the default in polars 😇)

**Review by MarcoGorelli — APPROVED** [review]
absolutely amazing, my hat is raised above my head 🎩

File: narwhals/_arrow/series.py (hunk 1 above)
Comment (FBruzzesi) [reply narwhals/_arrow/series.py:480]: I was planning to address that to make the option available (soon...)

**Pull request merged**